### DISCUSSION TITLE
Multiple image tag without applying git.

### DISCUSSION THREAD

**Discussion by adim** [fluxcd/flux2#2879] [Q&A] [answered]
Hi members,
I just finished setup flux2 my stack:
- k8s cluster on GCP 
- GitHub mono repo.
- artifact registry for image container & internal helm chart template.
- flux components: helm-controller, kustomize-controller,notification-controller,source-controller.

I have a base docker image for all web apps, each web app is workload inside the cluster (setup as helmRelease).
When the base docker image is changing GCP cloud build creates a new image in the repo, now all apps need to pull the new image.
flux gives 2 options to update all web apps:
1. Create pub/sub to call notification-controller.  issues:
-  require a domain/ssl certificate for HTTPS subscription(Google limitation). 
-  it's required to configure "ImageUpdateAutomation" in order to push a commit to a branch with a new image version, I prefer to avoid it because of git conflicts/ merage

2. Call "reconcile" for all apps.  issue: getting limit error.

Does flux have more options? Or maybe you think about another way to implement it?

Thanks.

**Comment by stefanprodan** [comment, marked as answer]
If you push a new Helm chart, you can set a semver range in Flux `HelmRelease` and it will do an automated upgrade. You can call `flux reconcile source helm` only once and Flux will know which chart changed.

**Reply by adim** [reply]
That is a great idea.
Right now all web apps use a single helm chart from flux-system.
So I can create a chart per namespace OR control chart name prefix by a substitute (namespace-name)

**Reply by adim** [reply]
I tried your solution (flux reconcile source helm).
It worked, all web-apps workloads got the new version of the chart (I see it in workload labels), but web-apps containers are still the old version, probably because I set the image tag as "latest".
Any option to tell Flux to force the cluster to pull a new container image?

**Reply by stefanprodan** [reply]
Flux doesn’t support mutable tags like latest because it can’t detect changes, something must change for a rollout to happen.

**Reply by adim** [reply]
I see, Is it possible to use ImageUpdateAutomation only for rollout? (without pushing the new version id to git)

**Reply by stefanprodan** [reply]
No, Flux is a GitOps tool, it will not change anything in-cluster unless it’s specified in Git.
